benchexec/tools/ultimate.py

```diff
@@ -71,15 +73,13 @@ def project_url(self):
 
     def executable(self, tool_locator):
         exe = tool_locator.find_executable("Ultimate.py")
-        dir_name = os.path.dirname(exe)
-        logging.debug("Looking in %s for Ultimate and plugins/", dir_name)
-        for _, dir_names, file_names in os.walk(dir_name):
-            if "Ultimate" in file_names and "plugins" in dir_names:
-                return exe
-            break
+        dir_name = Path(os.path.dirname(exe))
+        logging.debug(f"Checking if {dir_name} contains a launcher jar")
```

Comment:
We prefer `%s` for logging because the logging library has builtin support for that and can handle it lazily.

Reply:
:+1: fixed